account_bank_statement_import/models/account_bank_statement_import.py

```diff
@@ -349,6 +349,15 @@ def _complete_statement(self, stmt_vals, journal_id, account_number):
                         bank_account_id = bank_obj and bank_obj.id or False
                 line_vals['partner_id'] = partner_id
                 line_vals['bank_account_id'] = bank_account_id
+        if 'date' in stmt_vals and 'period_id' not in stmt_vals:
+            # if the parser found a date but didn't set a period for this date,
+            # do this now
+            try:
+                stmt_vals['period_id'] =\
+                    self.env['account.period'].find(dt=stmt_vals['date']).id
```

Comment:
According to https://github.com/odoo/odoo/blob/8.0/addons/account/account.py#L1042, you should put account_period_prefer_normal=True in the context to avoid obtaining an opening period.